src/renderer/components/NewServerModal.tsx

```diff
@@ -93,108 +78,171 @@ class NewServerModal extends React.PureComponent<Props, State> {
         });
     }
 
-    getServerUrlValidationError() {
-        if (!this.state.saveStarted) {
+    handleServerUrlChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const serverUrl = e.target.value;
+        this.setState({serverUrl, validationResult: undefined});
+        this.validateServerURL(serverUrl);
+    }
+
+    validateServerURL = (serverUrl: string) => {
+        clearTimeout(this.validationTimeout as unknown as number);
+        this.validationTimeout = setTimeout(() => {
+            this.setState({validationStarted: true});
+            window.desktop.validateServerURL(serverUrl, this.props.server?.id).then((validationResult) => {
```

Comment:
Similar issues to the ones raised in the functional version.

Reply:
Still potential issues around `setState` being called when the component is already unmounted.